File: server/router/DCSSMTConsumeAndReject.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitDCSSMTConsumeAndRejectRouter(Router *gin.RouterGroup) {
	DCSSMTConsumeAndRejectRouter := Router.Group("DSCR").Use(middleware.OperationRecord())
	{
		DCSSMTConsumeAndRejectRouter.POST("createDCSSMTConsumeAndReject", v1.CreateDCSSMTConsumeAndReject)   // 新建DCSSMTConsumeAndReject
		DCSSMTConsumeAndRejectRouter.DELETE("deleteDCSSMTConsumeAndReject", v1.DeleteDCSSMTConsumeAndReject) // 删除DCSSMTConsumeAndReject
		DCSSMTConsumeAndRejectRouter.DELETE("deleteDCSSMTConsumeAndRejectByIds", v1.DeleteDCSSMTConsumeAndRejectByIds) // 批量删除DCSSMTConsumeAndReject
		DCSSMTConsumeAndRejectRouter.PUT("updateDCSSMTConsumeAndReject", v1.UpdateDCSSMTConsumeAndReject)    // 更新DCSSMTConsumeAndReject
		DCSSMTConsumeAndRejectRouter.GET("findDCSSMTConsumeAndReject", v1.FindDCSSMTConsumeAndReject)        // 根据ID获取DCSSMTConsumeAndReject
		DCSSMTConsumeAndRejectRouter.GET("getDCSSMTConsumeAndRejectList", v1.GetDCSSMTConsumeAndRejectList)  // 获取DCSSMTConsumeAndReject列表
	}
}

File: web/src/api/DCSSMTConsumeAndReject.js
import service from '@/utils/request'

// @Tags DCSSMTConsumeAndReject
// @Summary 创建DCSSMTConsumeAndReject
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DCSSMTConsumeAndReject true "创建DCSSMTConsumeAndReject"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /DSCR/createDCSSMTConsumeAndReject [post]
export const createDCSSMTConsumeAndReject = (data) => {
     return service({
         url: "/DSCR/createDCSSMTConsumeAndReject",
         method: 'post',
         data
     })
 }


// @Tags DCSSMTConsumeAndReject
// @Summary 删除DCSSMTConsumeAndReject
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DCSSMTConsumeAndReject true "删除DCSSMTConsumeAndReject"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /DSCR/deleteDCSSMTConsumeAndReject [delete]
 export const deleteDCSSMTConsumeAndReject = (data) => {
     return service({
         url: "/DSCR/deleteDCSSMTConsumeAndReject",
         method: 'delete',
         data
     })
 }

// @Tags DCSSMTConsumeAndReject
// @Summary 删除DCSSMTConsumeAndReject
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除DCSSMTConsumeAndReject"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /DSCR/deleteDCSSMTConsumeAndReject [delete]
 export const deleteDCSSMTConsumeAndRejectByIds = (data) => {
     return service({
         url: "/DSCR/deleteDCSSMTConsumeAndRejectByIds",
         method: 'delete',
         data
     })
 }

// @Tags DCSSMTConsumeAndReject
// @Summary 更新DCSSMTConsumeAndReject
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DCSSMTConsumeAndReject true "更新DCSSMTConsumeAndReject"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /DSCR/updateDCSSMTConsumeAndReject [put]
 export const updateDCSSMTConsumeAndReject = (data) => {
     return service({
         url: "/DSCR/updateDCSSMTConsumeAndReject",
         method: 'put',
         data
     })
 }


// @Tags DCSSMTConsumeAndReject
// @Summary 用id查询DCSSMTConsumeAndReject
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DCSSMTConsumeAndReject true "用id查询DCSSMTConsumeAndReject"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /DSCR/findDCSSMTConsumeAndReject [get]
 export const findDCSSMTConsumeAndReject = (params) => {
     return service({
         url: "/DSCR/findDCSSMTConsumeAndReject",
         method: 'get',
         params
     })
 }


// @Tags DCSSMTConsumeAndReject
// @Summary 分页获取DCSSMTConsumeAndReject列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取DCSSMTConsumeAndReject列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /DSCR/getDCSSMTConsumeAndRejectList [get]
 export const getDCSSMTConsumeAndRejectList = (params) => {
     return service({
         url: "/DSCR/getDCSSMTConsumeAndRejectList",
         method: 'get',
         params
     })
 }